libraries/package-deps-hash/src/getRepoState.ts

```diff
@@ -394,11 +293,23 @@ export function getRepoState(currentWorkingDirectory: string, gitPath?: string):
     throw new Error(`git ls-tree exited with status ${lsTreeResult.status}: ${lsTreeResult.stderr}`);
   }
 
-  const state: Map<string, string> = parseGitLsTree(lsTreeResult.stdout);
+  const { files, submodules } = parseGitLsTree(lsTreeResult.stdout);
 
-  applySubmodulesTreeState(rootDirectory, state, gitPath);
+  applyWorkingTreeState(rootDirectory, files, gitPath);
 
-  applyWorkingTreeState(rootDirectory, state, gitPath);
+  const state: Map<string, string> = files;
+
+  // Existence check for the .gitmodules file
+  const hasSubmodules: boolean = FileSystem.exists(`${rootDirectory}/.gitmodules`);
```

Comment:
```suggestion
  const hasSubmodules: boolean = submodules.size > 0 && FileSystem.exists(`${rootDirectory}/.gitmodules`);
```
That way we can skip the file system access if we don't think we have submodules. That said, the FileSystem check might be redundant unless we think we can have objects of type `commit` that aren't submodules.